app/post/controllers/post-messages-controller.js

```diff
@@ -0,0 +1,210 @@
+module.exports = [
+    '$scope',
+    '$rootScope',
+    '$translate',
+    'post',
+    'ContactEndpoint',
+    'CollectionEndpoint',
+    'MessageEndpoint',
+    'Notify',
+    'UserEndpoint',
+    'PostEndpoint',
+    '$filter',
+    '_',
+function (
+    $scope,
+    $rootScope,
+    $translate,
+    post,
+    ContactEndpoint,
+    CollectionEndpoint,
+    MessageEndpoint,
+    Notify,
+    UserEndpoint,
+    PostEndpoint,
+    $filter,
+    _
+) {
+
+    $scope.post = post;
+    $scope.publishedFor = function () {
+        if ($scope.post.status === 'draft') {
+            return 'post.publish_for_you';
+        }
+        if (!_.isEmpty($scope.post.published_to)) {
+            return $scope.post.published_to.join(', ');
+        }
+
+        return 'post.publish_for_everyone';
+    };
+
+    if (post.contact.id) {
+        MessageEndpoint.allInThread({ contact: post.contact.id })
+            .$promise.then(function (messages) {
+                $scope.messages = messages;
+                _.each(messages, function (message, index) {
+                    if (message.user) {
+                        message.user = UserEndpoint.get({id: message.user.id});
+                    }
+                });
+            });
+    }
+
+    // Set the page title
+    $translate('post.message_thread').then(function (title) {
+        $scope.title = title;
+        $scope.$emit('setPageTitle', title);
+    });
+
+    ContactEndpoint.get({ id: $scope.post.contact.id })
```

Comment:
The original code of `$scope.contact = ContactEndpoint.get({ id: $scope.post.contact.id });` should have worked here.
Angular uses some magic to put the result back on the $scope.contact when its loaded..
Either works, so just an FYI